yt_dlp/extractor/youtube/_video.py

```diff
@@ -2832,10 +2840,14 @@ def fetch_po_token(self, client='web', context=_PoTokenContext.GVS, ytcfg=None,
         @param session_index: session index.
         @param player_url: player URL.
         @param video_id: video ID.
+        @param webpage: video webpage.
         @param kwargs: Additional arguments to pass down. May be more added in the future.
         @return: The fetched PO Token. None if it could not be fetched.
         """
 
+        # TODO(future): This validation should be moved into pot framework.
+        #  Some sort of middleware or validation provider perhaps?
+
```

Comment:
Would different providers want to override this? If not, keeping it here's better

Reply:
In the case for non-webpo tokens maybe (which the framework is designed to allow). But I don't see those being implemented anytime soon so it's not a priority to implement this.